django/db/models/indexes.py

```diff
@@ -0,0 +1,78 @@
+
+__all__ = ['Index']
+
+
+class BaseIndex(object):
+
+    # The type of index - BTree, Hash, RTree, etc.
+    # Used in deconstruction to differentiate indexes of different types with
+    # the same set of fields.
+    index_type = None
+
+    def __init__(self, *fields, **kwargs):
+        if not fields:
+            raise ValueError("Minimum one field required to define an index.")
+        self.fields = fields
+        if 'model' in kwargs:
+            self.model = kwargs['model']
+        if 'name' in kwargs:
+            self.name = kwargs['name']
```

Comment:
How can we ensure that we don't break other people's code even with this strategy. You are referring to code that has already been written by developers to create indexes (using `index_together`, `db_index`, etc.), am I right ?